### ISSUE TITLE
Not playing nice with dynamic data

### ISSUE THREAD

**Issue by philippeluickx** [AAkira/ExpandableLayout#24]
I tried to implement ExpandableLayout to my project, which resulted in following issue.

When I click the button to expand, I only get 1 line of text showing up (for a multi-line textview). This behaviour stays until I would rotate the device.
After rotation, everything works as expected.

I am using databinding, could have an effect? My impression is that the height of the relative layout is being calculated but not refreshed when new content arrives.

Hope this makes sense? Let me know if you need more info.

**Comment by AAkira** [comment]
Hi, @philippeluickx 

`ExpandableRelativeLayout` has the initialize method which is `initLayout(boolean)`.
It calculates a layout size again when it is called.
You call it if you want to initialize the layout.

Please tell me here if you cannot solve this problem.

**Comment by Mkrizanac** [comment]
I have the same issue, but i can't seem to find initLayout(boolean) method in ExpendableRelativeLayout, only initLayout() which does not help, if called,  that method sets the size to 0-

**Comment by AAkira** [comment]
@Mkrizanac 

I'm sorry. I added it on v1.4.0.
Try this.
`compile 'com.github.aakira:expandable-layout:1.4.0@aar'`

**Comment by Mkrizanac** [comment]
@AAkira 

hmm, weird, i can see the method on git:
```@Override
    public void initLayout(final boolean isMaintain) {
        closePosition = 0;
        layoutSize = 0;
        isArranged = isMaintain;
        isCalculatedSize = false;
        savedState = null;

        super.requestLayout();
    }```

and i updated to 1.4.0, but still, the only method i can see in my ExpandableRelativeLayout is
![image](https://cloud.githubusercontent.com/assets/6604210/10943008/cde303ce-8312-11e5-9315-24954c09ceaf.png)

**Comment by AAkira** [comment]
@Mkrizanac 

Thank you for reporting.
I upload the same version which is v1.4.0 again.
Please check it.

**Comment by Mkrizanac** [comment]
@AAkira 

thanks, i can use that method now, but it's not helping, when i call it i can't even expand my layout anymore.

**Comment by AAkira** [comment]
@Mkrizanac 

Yes, you are right.
You should call `initLayout()` every time when the size is changed.
This library supports that the layout has the size decided in advance.
Or you can use `move(position)` method which can moves expandable layout to optional position.

**Comment by Mkrizanac** [comment]
@AAkira 

still not working, size is set to 0 whenever i call initLayout(boolean) and it's not changed. 
I ended up doing simple workaround, i always get all the data before opening fragment so once fragment with expandable layout starts, it already contains all the content needed to show, that way, size get's calculated correctly (if the data is set in onCreateView).  Problem still happens when changing content, which could be solved by creating fragment again, which is not a very good solution.

**Comment by AAkira** [comment]
@Mkrizanac 

Yes, this library supports that it already contains all the content needed to show.
So, you should call `initLayout(boolean)` which calculates the layout size again if the layout size changes.
Can you show me your code here?

**Comment by philippeluickx** [comment]
Doesn't work for me either. I call the initlayout(true) on my onLoadFinished method, but then nothing changes. Without calling it, at least it bounced with 1 line...

**Comment by antoxa2584x** [comment]
Same issue when wrap content view inside changes it size

**Comment by anirudhmaddy** [comment]
Call initLayout(boolean) before adding dynamic layout. This solved my problem.

**Comment by DaRolla** [comment]
in my case #49 it didn't help

**Comment by AAkira** [comment]
@philippeluickx 
@Mkrizanac 
@antoxa2584x 
@anirudhmaddy 
@DaRolla 

I'm sorry to reply late.
I released v1.5.0.
Please check it.

**Comment by AAkira** [comment]
Hi, there
I released v1.6.0.
This is fixed in v1.6.0.

**Comment by derlysbenito** [comment]
I find your library fascinating, I'm implemented now but I have a problem using a recycler with the ExplandableRelativeLayout, it happens that I can not hide or show my recycler, use toogle methods, move but I do not get results, I hope you can help me.
